api/envoy/extensions/filters/http/lua/v3/lua.proto

```diff
@@ -24,4 +26,26 @@ message Lua {
   // be properly escaped. YAML configuration may be easier to read since YAML supports multi-line
   // strings so complex scripts can be easily expressed inline in the configuration.
   string inline_code = 1 [(validate.rules).string = {min_bytes: 1}];
+
+  // The Lua source codes that Envoy will execute based on name. Key is Lua source code name, e.g.
+  // ``foo.lua``, this name can be referenced by
+  // :ref:`LuaPerRoute.name <envoy_v3_api_msg_extensions.filters.http.lua.v3.LuaPerRoute.name>`.
+  map<string, config.core.v3.DataSource> source_codes = 2;
+}
+
+message LuaPerRoute {
+  oneof override {
+    option (validate.required) = true;
+
+    // Disable the Lua filter for this particular vhost or route. If disabled is specified in
+    // multiple per-filter-configs, the most specific one will be used.
+    bool disabled = 1 [(validate.rules).bool = {const: true}];
+
+    // The name of a Lua source code defined in
+    // :ref:`Lua.source_codes <envoy_v3_api_msg_extensions.filters.http.lua.v3.Lua.source_codes>`.
+    string name = 2 [(validate.rules).string = {min_bytes: 1}];
+
+    // Lua source code from RDS directly.
+    config.core.v3.DataSource code = 3;
```

Comment:
Should `name` and `code` be in a `oneof`?

Reply:
The `name` is used to refer to a Lua script in the global configuration, and the `code` is used to configure a Lua script directly in the Route. They should not be configured repeatedly.

Reply:
We cannot set the `name` and `code` at the same time. If you need to use the same script in different routes, you can send the script through LDS, and then set the `name` field in the route configuration. When the script is large, a part of redundant configuration can be reduced. The `code` is for better flexibility, to avoid all scripts must be issued using LDS.

Reply:
I suggest a `oneof` for this.

Reply:
OK, it seems like it is. One of `disabled`, `name`, or `code`.

Reply:
Looks good, but nit, rename s/code/source_code/ for consistency with `source_codes`.

Reply:
@wbpcode could you rename this to `source_code` as per @htuch's comment?